skopt/tests/test_optimizer.py

```diff
@@ -89,3 +91,21 @@ def test_returns_result_object():
     assert isinstance(result, OptimizeResult)
     assert_equal(len(result.x_iters), len(result.func_vals))
     assert_equal(np.min(result.func_vals), result.fun)
+
+param_list = [(ExtraTreesRegressor(random_state=2), 'lbfgs'),
+              (RandomForestRegressor(random_state=2), 'lbfgs')]
+def test_acq_optimizer_generator():
+    for params in param_list:
+        yield check_acq_optimizer, params[0], params[1]
+
+@raises(ValueError)
+def check_acq_optimizer(base_estimator, acq_optimizer):
+    opt = Optimizer([(-2.0, 2.0)], base_estimator=base_estimator,
+                    n_random_starts=1, acq_optimizer=acq_optimizer)
+
+def test_acq_optimizer():
+    base_estimator = ExtraTreesRegressor(random_state=2)
+    opt = Optimizer([(-2.0, 2.0)], base_estimator, n_random_starts=1,
+                    acq_optimizer="auto")
+
+    assert opt.acq_optimizer == "sampling"
```

Comment:
I had meant not to modify `opt.acq_optimizer`. You can set `opt.acq_optimizer_` (with the underscore) and use that attribute wherever you want.